server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -2299,6 +2298,15 @@ public void addFlushListener(Translog.Location location, ActionListener<Long> li
         listener.onFailure(new UnsupportedOperationException("Engine type " + this.getClass() + " does not support flush listeners."));
     }
 
+    /**
+     * Safely close the provided engine
+     */
+    public static void close(@Nullable Engine engine) throws IOException {
+        if (engine != null) {
+            engine.close();
+        }
+    }
```

Comment:
Is this one just a sort-of null-safe close, or is the idea to abstract over how engines are closed? Perhaps the javadoc should explain that and the blocking-ness (or what the caller can expect when it returns).

Reply:
Also if it's blocking and that's something we're in the process of moving away from, should it perhaps live in test scope?